equinox-shipment/Domain/Container.fs

```diff
@@ -0,0 +1,56 @@
+module Container
+
+type Container = { id: string; shipmentIds: string[]; finalized: bool }
+
+module Events =
+
+    let [<Literal>] CategoryId = "Shipment"
+
+    let (|ForClientId|) (clientId: string) = FsCodec.StreamName.create CategoryId clientId
+
+    type Event =
+        | ContainerCreated    of containerId  : string
+        | ContainerFinalized  of shipmentIds  : string[]
+        | Snapshotted         of Container
+        interface TypeShape.UnionContract.IUnionContract
+
+    let codec = FsCodec.NewtonsoftJson.Codec.Create<Event>()
+
+module Fold =
+
+    type State = Container
+
+    let initial: State = { id = null; shipmentIds = [||]; finalized = false }
+
+    let evolve (state: State) (event: Events.Event): State =
+        match event with
+        | Events.Snapshotted        snapshot    -> snapshot
+        | Events.ContainerCreated   containerId -> { state with id = containerId }
+        | Events.ContainerFinalized shipmentIds -> { state with shipmentIds = shipmentIds }
+
+
+    let fold: State -> Events.Event seq -> State =
+        Seq.fold evolve
+
+    let isOrigin (event: Events.Event) =
+        match event with
+        | Events.Snapshotted _ -> true
+        | _ -> false
+
+    let snapshot state = Events.Snapshotted state
```

Comment:
it can be useful to add `(state : State)` to get early warnings in IDE if the Fold State type varies from the event type

Reply:
Updated thanks!